sdk/rpc/proto/protocol.go

```diff
@@ -27,18 +27,32 @@ func (c CredentialID) String() string {
 	return fmt.Sprintf("plugin.Credentials[%d]", c)
 }
 
+// CredentialUsageID uniquely identifies a CredentialUsage within a plugin.
+type CredentialUsageID struct {
+	Executable ExecutableID
+	Usage      int
+}
+
+func (c CredentialUsageID) String() string {
+	return fmt.Sprintf("%s.Uses[%d]", c.Executable, c.Usage)
+}
+
 // ProvisionerID uniquely identifies a provisioner within a plugin.
 type ProvisionerID struct {
-	Plugin     string
-	Credential sdk.CredentialName
-	Executable *ExecutableID
+	// IsCredentialProvisioner is set to true if the ProvisionerID identifies the DefaultProvisioner of a credential.
+	// It is set to false if the ProvisionerID identifies the Provisioner of an executable's CredentialUsage.
+	IsCredentialProvisioner bool
```

Comment:
Technically all provisioners are "credential provisioners". Therefore, I'd say we name this `IsDefaultProvisioner`

Reply:
Makes sense :+1: